ColorDeductoActivity.py

```diff
@@ -84,7 +73,12 @@ def __init__(self, handle):
 
         self._sharing = False
         self._initiating = False
-        self._setup_presence_service()
+        self.connect('shared', self._shared_cb)
+        self.connect('joined', self._joined_cb)
+        self._collab = CollabWrapper(self)
+        self._collab.connect('message', self._message_cb)
+        self._collab.connect('joined', self._joined_cb)
```

Comment:
Can you keep the signals in the same stack? Why connect both from sugar and collabwrapper to the same callbacks?

Reply:
Done in 2d5baeb.

Reply:
Thanks.  They are the same signal name, but they have different meaning, so they aren't the same callbacks.  One indicates readiness as far as sugar3.activity is concerned, the other indicates as far as CollabWrapper is concerned.

Reply:
Yes both signal names have different meaning and get invoked differently but the same callback gets executed when either is invoked.